Completes the description of community provider's lifecycle (#35277)
mistercrunch/Airflow: PROVIDERS.rst

themselves when they own, manage and release their provider, especially when they can synchronize releases
of their provider with new feature, the service might get added.

Community providers release process
-----------------------------------

The community providers are released regularly (usually every 2 weeks) in batches consisting of any providers
that need to be released because they changed since last release. The release manager decides which providers
to include and whether some or all providers should be released (see the next chapter about upgrading the

> minimum version of Airflow for example of case where we release all active, not suspended providers,

Reviewer: ```suggestion
minimum version of Airflow for example the case where we release all active meaning non-suspended providers,
```

Increasing the minimum version ot the Providers is one of the reasons why 3rd-party provider maintainers
might want to maintain their own providers - as they can decide to support older versions of Airflow.

3rd-parties relation to community providers
-------------------------------------------

> Providers, can (and it is recommended for 3rd-party services) also be maintained and releases by 3rd parties,

Reviewer: ```suggestion
Providers, can (and it is recommended for 3rd-party services) also be maintained and released by 3rd parties,
```

but for multiple reasons we might decide to keep those providers as community managed providers - mostly
due to prevalence and popularity of the 3rd-party services and use cases they serve among our community. There

> are however certain condition and expectations we have in order

Reviewer: ```suggestion
are however certain conditions and expectations we have in order.
```

where regular rules for code modification apply (following the
`Apache Software Foundation voting rules <https://www.apache.org/foundation/voting.html#votes-on-code-modification>`_).
In cases where the reasons for removal are ``obvious``, and discussed before, also ``[LAZY CONSENSUS]`` thread
can be started. Generally speaking a discussion thread ``[DISCUSS]`` is advised before such removal and
sufficient time should pass (at least a week) to give a chance for community members to express their
opinion on the removal.

There are the following consequences (or lack of them) of removing the provider:

* One last release of the provider is done with documentation updated informing that the provider is no
  longer maintained by the Apache Airflow community - linking to this page. This information should also
  find its way to the package documentation and consequently - to the description of the package in PyPI.
* An ``[ANNOUNCE]`` thread is sent to the devlist and user list announcing removal of the provider
* The released provider packages remain available on PyPI and in the Downloads/Archives of the Apache

>   Software Foundation, also it remains in the Index of the Apache Airflow Providers at

Reviewer 2: Do we keep it in svn?

Author: Good point.

Indeed SVN finally gets to "Downloads/Archive": https://downloads.apache.org/airflow/ you can see latest release there and https://archive.apache.org/dist/airflow/  include all historical releases. 

We cannot remove things from archives (and that's fine), but indeed we can (and probably should) releave it from SVN which will remove it from "downloads".

Author: Updated/Clarified.

> * The released provider packages remain available on PyPI and in the Downloads/Archives of the Apache
>   Software Foundation, also it remains in the Index of the Apache Airflow Providers at
>   `Airflow Documentation <https://airflow.apache.org/docs/>`_ with note ``(not maintained)`` next to it.

Author: ```suggestion
* The released provider packages remain available on PyPI and in the
   `Archives <https://archive.apache.org/dist/airflow/providers/>`_ of the Apache
   Software Foundation, while they are removed from the 
   `Downlads <https://downloads.apache.org/airflow/providers/>`_ .
   Also it remains in the Index of the Apache Airflow Providers documentation at
   `Airflow Documentation <https://airflow.apache.org/docs/>`_ with note ``(not maintained)`` next to it.
```

The suspension will be lifted when the dependencies of the provider are made compatible with the Apache
Airflow and with other providers - by merging a PR that removes the suspension and succeeds.

Removing community providers
----------------------------

The providers can be removed from main branch of Airflow when the community agrees that there should be no
more updates to the providers done by the community - except maybe potentially security fixes found. There
might be various reasons for the providers to be removed:

* the service they connect to is no longer available
* the dependencies for the provider are not maintained any more and there is no viable alternative
* there is another, more popular provider that supersedes community provider

> * etc. etc.

Reviewer 3: 😉 💯

where regular rules for code modification apply (following the
`Apache Software Foundation voting rules <https://www.apache.org/foundation/voting.html#votes-on-code-modification>`_).
In cases where the reasons for removal are ``obvious``, and discussed before, also ``[LAZY CONSENSUS]`` thread
can be started. Generally speaking a discussion thread ``[DISCUSS]`` is advised before such removal and
sufficient time should pass (at least a week) to give a chance for community members to express their
opinion on the removal.

There are the following consequences (or lack of them) of removing the provider:

* One last release of the provider is done with documentation updated informing that the provider is no
  longer maintained by the Apache Airflow community - linking to this page. This information should also
  find its way to the package documentation and consequently - to the description of the package in PyPI.
* An ``[ANNOUNCE]`` thread is sent to the devlist and user list announcing removal of the provider
* The released provider packages remain available on PyPI and in the
   `Archives <https://archive.apache.org/dist/airflow/providers/>`_ of the Apache

>    Software Foundation, while they are removed from the 

Author: ```suggestion
   Software Foundation, while they are removed from the
```

>    `Downlads <https://downloads.apache.org/airflow/providers/>`_ .

Reviewer 4: ```suggestion
   `Downloads <https://downloads.apache.org/airflow/providers/>`_ .
```
